### PR TITLE
Bug 1932152: UPSTREAM: 809: Cap IOPS when calculating from iopsPerGB

### PR DESCRIPTION
**Pull request by jsafrane**
Consider AWS IOPS limits when calculating IOPS from iopsPerGB to always
provision a supported volume. Users should be create a StorageClass "fast"
that provides max. IOPS that AWS can support for any volume size they
provide, without errors.

In the end, this PR provides much nicer messages to users when they use wrong `iopsPerGB` / volume size combination.

@openshift/storage

### CODE REVIEW

File: pkg/cloud/cloud.go [hunk 1]
```diff
@@ -1073,3 +1095,29 @@ func getVolumeAttachmentsList(volume *ec2.Volume) []string {
...
+		}
+	}
+	if iops > maxTotalIOPS {
+		iops = maxTotalIOPS
+		klog.V(5).Infof("Capped IOPS for %s %d GB volume at the max supported limit: %d", volumeType, requestedCapacityGiB, iops)
+	}
+	if iops > maxIOPSPerGB*requestedCapacityGiB {
+		iops = maxIOPSPerGB * requestedCapacityGiB
+		klog.V(5).Infof("Capped IOPS for %s %d GB volume at %d IOPS/GB: %d", volumeType, requestedCapacityGiB, maxIOPSPerGB, iops)
+	}
```
Comment (Elbehery) [inline pkg/cloud/cloud.go:1121]: idiomatic go 👍🏽

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**